### PR TITLE
chore: avoid use default mux

### PR DESCRIPTION
**Pull request by jim3ma**
## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation Update (if none of the other choices apply)

## Checklist

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.

### CODE REVIEW

File: client/daemon/proxy/proxy.go [hunk 1]
```diff
@@ -290,9 +271,26 @@ func NewProxy(options ...Option) (*Proxy, error) {
...
+	proxy.updateMirrorHandler()
 	return proxy, nil
 }
 
+func (proxy *Proxy) updateMirrorHandler() {
+	h := proxy.directHandler
+	if proxy.registry == nil || proxy.registry.Remote == nil || proxy.registry.Remote.URL == nil {
+		logger.Warnf("registry mirror url is empty, registry mirror feature is disabled")
+		h.HandleFunc("/", func(w http.ResponseWriter, r *http.Request) {
+			http.Error(w, fmt.Sprintf("registry mirror feature is disabled"), http.StatusNotFound)
```
Comment (aobt) [inline client/daemon/proxy/proxy.go:285]: 这里可以顺便去掉不必要的 fmt.Sprintf

Comment (jim3ma) [reply client/daemon/proxy/proxy.go]: Done

**Review by aobt — APPROVED** [review]
LGTM

**Review by gaius-qi — APPROVED** [review]
LGTM

**Review by gaius-qi — APPROVED** [review]
LGTM

**Pull request merged**